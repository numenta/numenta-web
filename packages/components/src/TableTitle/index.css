/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableTitle {
  composes: bg-near-white from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: v-top from 'tachyons';
}
