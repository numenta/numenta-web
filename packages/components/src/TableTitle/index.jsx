// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


/**
 *
 */
const TableTitle = ({border, children, colSpan}) => {
  const classes = [styles.tableTitle]

  if (border) classes.push(styles.border)

  return (
    <th className={classNames(...classes)} colSpan={colSpan}>
      {children}
    </th>
  )
}

TableTitle.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node,
  colSpan: React.PropTypes.number.isRequired,
}

TableTitle.defaultProps = {
  colSpan: 1,
}

export default TableTitle
