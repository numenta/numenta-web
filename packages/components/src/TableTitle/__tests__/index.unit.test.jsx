import React from 'react'
import renderer from 'react-test-renderer'

import TableTitle from '../../TableTitle'


describe('TableTitle React component', () => {

  it('Renders Bordered correctly', () => {
    const component = renderer.create(
      <TableTitle border={true}>
        Some Statistics
      </TableTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Un-bordered correctly', () => {
    const component = renderer.create(
      <TableTitle border={false}>
        Some Statistics
      </TableTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders with Expanded Column Span correctly', () => {
    const component = renderer.create(
      <TableTitle colSpan={2}>
        Some Statistics
      </TableTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
