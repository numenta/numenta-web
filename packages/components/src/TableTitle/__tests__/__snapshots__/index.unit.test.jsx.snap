exports[`TableTitle React component Renders Bordered correctly 1`] = `
<th
  className="tableTitle border"
  colSpan={1}>
  Some Statistics
</th>
`;

exports[`TableTitle React component Renders Un-bordered correctly 1`] = `
<th
  className="tableTitle"
  colSpan={1}>
  Some Statistics
</th>
`;

exports[`TableTitle React component Renders with Expanded Column Span correctly 1`] = `
<th
  className="tableTitle"
  colSpan={2}>
  Some Statistics
</th>
`;
